src/transform/server.js

```diff
@@ -0,0 +1,215 @@
+      return t.spreadProperty(attr.argument);
+    }
+
+    const { name, value } = attr;
+    return t.objectProperty(
+      t.identifier(name.name),
+      t.isJSXExpressionContainer(value) ?
+        value.expression :
+        value
+    );
+  });
+
+const pushVanillaVdom = (segments, node) => {
+  const { openingElement, children, closingElement } = node;
+
+  segments.push(`<${openingElement.name.name}`);
+  pushAttributes(segments, openingElement.attributes);
+  if (openingElement.selfClosing) {
+    segments.push(`></${openingElement.name.name}>`);
+    return;
+  }
+
+  segments.push(">");
+
+  children.forEach(child => {
+    if (t.isJSXText(child)) {
+      const trimmed = child.value.trim();
+      if (trimmed) { segments.push(htmlStringEscape(trimmed)); }
+    } else if (t.isJSXExpressionContainer(child)) {
+      segments.push(
+        buildObjectExpression({
+          __prerendered__: t.stringLiteral("expression"),
+          expression: child.expression
+        })
+      );
+    } else {
+      segments.push(child);
+    }
+  });
+
+  segments.push(`</${closingElement.name.name}>`);
+};
+
+const pushAttributes = (segments, attrs) => {
+  attrs.forEach(attr => {
+    const { name: { name }, value } = attr;
+
+    if (!value) {
+      // If no value is specified, this is a boolean HTML attribute that is set.
+      segments.push(` ${name}`);
+    } else if (t.isStringLiteral(value)) {
+      // If the value is a string literal, it is pre-renderable.
+      segments.push(` ${name}="${value.value}"`);
+    } else {
+      segments.push(
+        buildObjectExpression({
+          __prerendered__: t.stringLiteral("attr"),
+          attrObj: buildObjectExpression({
+            [name]: value.expression
+          })
+        })
+      );
+    }
+  });
+  segments.push(
+    // This is used by the renderer to insert `data-react-id`s.
+    t.numericLiteral(REACT_ID)
+  );
+};
+
+const compress = elements => {
+  return elements.reduce((memo, element) => {
+    const prevIdx = memo.length - 1;
+    const prev = memo[prevIdx];
+
+    if (t.isStringLiteral(prev) && t.isStringLiteral(element)) {
+      prev.value += element.value;
+    } else {
+      memo.push(element);
+    }
+
+    return memo;
+  }, []);
+};
+
+const expressionifyStringSegments = segments => {
+  return segments.map(segment =>
+    typeof segment === "string" ?
+      t.stringLiteral(segment) :
+      segment
+  );
+};
+
+const buildObjectExpression = obj => {
+  return t.objectExpression(Object.keys(obj).map(key => {
+    return t.objectProperty(
+      t.identifier(key),
+      obj[key]
+    );
+  }));
+};
+
+const objectExpressionToObject = objExpr => {
+  const obj = Object.create(null);
+  objExpr.properties.forEach(property => {
+    obj[property.key.name || property.key.value] = property.value;
+  });
+  return obj;
+};
+
+const isHoistable = arrayElements => {
+  return arrayElements.reduce((memo, el) => {
+    if (!memo) { return memo; }
+    if (
```

Comment:
Optional -- Instead of;

```js
if (LONG_CONDITION) {
  return true;
}
return false;
```

You could simplify to:

```js
return LONG_CONDITION;
```